eclipse-collections/src/main/java/org/eclipse/collections/impl/list/primitive/IntInterval.java

```diff
@@ -1007,4 +1016,33 @@ public int next()
             throw new NoSuchElementException();
         }
     }
+
+    private class IntIntervalSpliterator implements Spliterator.OfInt {
+        AtomicInteger current = new AtomicInteger(from);
```

Comment:
It's been years since I looked at the internals so correct me if my assumptions are wrong here. I believe that spliterators form the core of the fork/join tasks submitted to a pool. I believe that each task will either split or iterate. More specifically, I don't think the parallel algorithms ever advance a spliterator before splitting it. Because of this, I think there is a clean separation between two kinds of fields. Boundaries like startIndex and endIndex will be final and can freely be used for communication between tasks. State like currentIndex will be mutable but only read/written by a single thread.

Reply:
@motlin I agree with your description.